localstack/services/events/target.py

```diff
@@ -0,0 +1,317 @@
+        raise NotImplementedError("Sagemaker target is not yet implemented")
+
+
+class SnsTargetService(TargetService):
+    def send_event(self, event):
+        sns_client = self.client.sns.request_metadata(
+            service_principal=self.service, source_arn=self.rule_arn
+        )
+        sns_client.publish(TopicArn=self.target["Arn"], Message=json.dumps(event))
+
+
+class SqsTargetService(TargetService):
+    def send_event(self, event):
+        sqs_client = self.client.sqs.request_metadata(
+            service_principal=self.service, source_arn=self.rule_arn
+        )
+        queue_url = sqs_queue_url_for_arn(self.target["Arn"])
+        msg_group_id = self.target.get("SqsParameters", {}).get("MessageGroupId", None)
+        kwargs = {"MessageGroupId": msg_group_id} if msg_group_id else {}
+        sqs_client.send_message(
+            QueueUrl=queue_url, MessageBody=json.dumps(event, separators=(",", ":")), **kwargs
+        )
+
+
+class StatesTargetService(TargetService):
+    """Step Functions Target Sender"""
+
+    def send_event(self, event):
+        stepfunctions_client = self.client.stepfunctions.request_metadata(
+            service_principal=self.service, source_arn=self.rule_arn
+        )
+        stepfunctions_client.start_execution(
+            stateMachineArn=self.target["Arn"], input=json.dumps(event)
+        )
+
+
+class SystemsManagerService(TargetService):
+    def send_event(self, event):
+        raise NotImplementedError("Systems Manager target is not yet implemented")
+
+    def _validate_input(self, target: Target):
+        super()._validate_input(target)
+        if not collections.get_safe(target, "$.RunCommandParameters.RunCommandTargets"):
+            raise ValueError(
+                "RunCommandParameters.RunCommandTargets is required for Systems Manager target"
+            )
+
+
+class TargetServiceFactory:
+    # supported targets: https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-targets.html
+    target_map = {
+        "apigateway": ApiGatewayTargetService,
+        "appsync": AppSyncTargetService,
+        "batch": BatchTargetService,
+        "ecs": ContainerTargetService,
+        "events": EventsTargetService,
+        "firehose": FirehoseTargetService,
+        "kinesis": KinesisTargetService,
+        "lambda": LambdaTargetService,
+        "logs": LogsTargetService,
+        "redshift": RedshiftTargetService,
+        "sns": SnsTargetService,
+        "sqs": SqsTargetService,
+        "sagemaker": SagemakerTargetService,
+        "ssm": SystemsManagerService,
+        # TODO custom endpoints via http target
+    }
+
+    def __init__(self, target: Target, region: str, account_id: str, rule_arn: Arn):
+        self.target = target
+        self.region = region
+        self.account_id = account_id
+        self.rule_arn = rule_arn
+
+    @staticmethod
+    def extract_service_from_arn(arn: Arn) -> str:
+        arn = parse_arn(arn)
+        return arn["service"]
```

Comment:
Seems to be identical with `extract_service_from_arn`

Reply:
true good catch, used the system method fixed in:https://github.com/localstack/localstack/pull/10613/commits/f872ce7be5c86eebbb54312045fb28fd635e23c9